src/function/arithmetic/multiply.js

```diff
@@ -420,6 +420,11 @@ export const createMultiply = /* #__PURE__ */ factory(name, dependencies, ({ typ
     // update ptr
     cptr[bcolumns] = cindex.length
 
+    // Set the type to undefined if it doesn't match
+    if (adt !== a._datatype || bdt !== b._datatype) {
```

Comment:
Can you also move this logic to like 376 where the matrix `c` is created and inline it?

Reply:
Done